### PR TITLE
GH Actions refactor

### PR DESCRIPTION
**Pull request by acasajus**
- Create docker container also on tag
- Only run the docker container creation once 
- Run the workflow on push and on PR creation so it's run only once when the PR is updated with new commits

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -93,8 +96,13 @@ jobs:
           name: code-coverage-report
           path: htmlcov
 
+  build:
+    runs-on: ubuntu-latest
+    needs: ['test']
+    if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v'))
```
Comment (nguyenkims) [inline .github/workflows/main.yml:102]: Do you know which docker image tag will be used here when the image is pushed?

**Review by nguyenkims — APPROVED** [review]
LGTM, just left a question.

File: .github/workflows/main.yml (hunk 1 above)
Comment (acasajus) [reply .github/workflows/main.yml:102]: Good question. I assumed that `tag_with_ref` would take the updated ref and use it. But looking at the docs there is no info about it and a quick google search seems it's deprecated [here](https://github.com/docker/build-push-action/issues/116). I'll update the action to use the newer data.

Comment (acasajus) [reply .github/workflows/main.yml:102]: Fixed updating to the lastest docker actions.

**Pull request merged**